rpmlint/checks/FileDigestCheck.py

```diff
@@ -34,46 +38,90 @@ def _get_digest_group_from_path(self, path):
                     pass
         return None
 
-    def check_binary(self, pkg):
+    def _in_digest_configuration_group(self, pkgfile):
+        if stat.S_ISDIR(pkgfile.mode):
+            return False
+        return self._get_digest_location_from_path(pkgfile.name)
+
+    def _check_symlinks(self, pkg):
         """
-        TODO: add comment
+        Check that all symlinks point to a correct location and that
+        symlinks are allowed in a configuration group.
         """
-        known_digests = self.package_digests.get(pkg.name)
+        result = True
         for filename, pkgfile in pkg.files.items():
-            group = self._get_digest_group_from_path(filename)
-            if not group:
+            group = self._get_digest_location_from_path(filename)
+            if not self._in_digest_configuration_group(pkgfile):
```

Comment:
this is a bit strange, we already know whether a group matches this filename (`group != None`), but then in `in_digest_configuration_group` we check that again. Maybe call this rather something like `_is_restricted_file(pkgfile`) and only check the file type in there.